OsmAnd/src/net/osmand/plus/search/dialogs/QuickSearchDialogFragment.java

```diff
@@ -348,7 +353,10 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
 		buttonToolbarMap.setOnClickListener(v -> {
 					cancelSearch();
 					SearchPhrase searchPhrase = searchUICore.getPhrase();
-					if (foundPartialLocation) {
+					PoiUIFilter searchListFilter = ((QuickSearchListAdapter)mainSearchFragment.getAdapter()).getPoiUIFilter();
```

Comment:
Can we get this filter inside  if (searchPhrase.isNoSelectedType() || searchPhrase.isLastWord(POI_TYPE)) ?